src/complexity_measures/statistical_complexity.jl

```diff
@@ -0,0 +1,192 @@
+```
+
+NOTE: I'm not sure if this API is congruent with the general complexity API, but I think
+that, since the "standard" signature is also possible, it should be fine.
+I have implemented the other signature because I find it redundant to calculate the entropy
+twice, which is needed for the complexity anyway and usually calculated in combination with it.
+
+Because the distributions of permutation symbols are needed both for the permutation entropy and the
+statistical complexity, the same can also be achieved using
+```julia
+p = probabilities(est, x)
+permutation_entropy = entropy(est, x)
+statistical_complexity = complexity(c, p; h=permutation_entropy)
+```
+
+[^Rosso2007] Rosso, O. A., Larrondo, H. A., Martin, M. T., Plastino, A., & Fuentes, M. A. (2007). 
+            [Distinguishing noise from chaos](https://doi.org/10.1103/PhysRevLett.99.154102).
+            Physical review letters, 99(15), 154102. 
+"""
+Base.@kwdef struct StatisticalComplexity <: ComplexityMeasure
+    est::ProbabilitiesEstimator = SymbolicPermutation()
+    distance::SemiMetric = JSDivergence()
+    entropy::Entropy = Renyi()
+end
+
+alphabet_length(c::StatisticalComplexity) = alphabet_length(c.est)
+
+function complexity(c::StatisticalComplexity, p::Probabilities; h::Float64) 
+    L = alphabet_length(c)
+    # fill with the probability distribution zeros for empty bins, 
+    # we need this to compare to a uniform distribution of m! patterns
+    p = Probabilities(cat(zeros(L-length(p)), p.p, dims=1))
+    # normalization is done automatically by Probabilities
+    uniform = Probabilities(ones(L))
+
+    dist = evaluate(c.distance, p, uniform)
+    dist *= h
+
+    return dist
+end
+
+function complexity(c::StatisticalComplexity, x::AbstractArray)
+    p = probabilities(x, c.est)
+    L = alphabet_length(c)
+    h = entropy(c.entropy, p) / log(c.entropy.base, L)
+    complexity(c, p; h)
+end
+
+function complexity_normalized(c::StatisticalComplexity, p::Probabilities; h::Float64)
+    comp = complexity(c, p; h)
+    L = alphabet_length(c)
+    uniform = Probabilities(ones(L))
+    # generate "deterministic" distribution (only one completely full bin)
+    determ = Probabilities(Base.append!(zeros(L-1), 1.0))
+    # distance to this distribution is the maximum possible distance to a uniform distribution
+    return comp / evaluate(c.distance, determ, uniform)
+end
+
+function complexity_normalized(c::StatisticalComplexity, x::AbstractArray)
+    comp = complexity(c, x)
+    L = alphabet_length(c)
+    uniform = Probabilities(ones(L))
+    # generate "deterministic" distribution (only one completely full bin)
+    determ = Probabilities(Base.append!(zeros(L-1), 1.0))
+    # distance to this distribution is the maximum possible distance to a uniform distribution
+    return comp / evaluate(c.distance, determ, uniform)
+end
+
+linearpermissiverange(start; stop, length) = length==1 ? (start:start) : range(start, stop=stop, length=length)
```

Comment:
This is a type unstable function, which means it brings terrible performance wherever it is used...

Reply:
You need to think of an alternative in how to go about this iteration.